website/docs/internals/json-format.mdx

```diff
@@ -629,8 +627,7 @@ A `<checks-representation>` describes the current state of a checkable object in
       // "kind" specifies what kind of checkable object this is. Different
       // kinds of object will have different additional properties inside the
       // address object, but all kinds include both "kind" and "to_display".
-      // Currently the two valid kinds are "resource" and "output_value", but
-      // additional kinds may be added in future Terraform versions.
+      // The two valid kinds are "resource" and "output_value".
```

Comment:
The fact that we have the overall warning that everything here is subject to change in future makes this moot right now, but I think this is a potential example of the category of things where it's worth being explicit that the set of valid values isn't closed because anyone developing against this format should take that into account and make sure their code will do something reasonable if it encounters a value it wasn't built to accept.

I wonder if there's a different way we can talk about enumerations that are open to future expansion without it sounding like a promise to do something. :thinking:

Reply:
yeah i need to think on this a bit. I see where you're coming from and I feel like what we need is official language to this effect. I'm going to make a ticket on my backlog so I don't forget about this.